core/parameters/src/config.rs

```diff
@@ -241,3 +252,30 @@ pub struct WitnessConfig {
     /// Soft size limit of storage proof used to validate new transactions in ChunkStateWitness.
     pub new_transactions_validation_state_size_soft_limit: usize,
 }
+
+/// Configures gas limit adjustments at runtime.
+///
+/// This functionality is not intended for production, but only for tests or benchmarks.
+///
+/// A possible use case is making a node that runs benchmarks automatically set the maximum gas
+/// limit that it can sustain.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct GasLimitAdjustmentConfig {
+    /// Override for `GAS_LIMIT_ADJUSTMENT_FACTOR`.
+    ///
+    /// The default adjustment factor might not be effective enough.
+    pub adjustment_factor_override: u64,
```

Comment:
I don't think you need an "_override" suffix here, it doesn't bring any more clarity and is inconsistent with other config values. I think it would make more sense if this field was of type `Option` and it was applied conditionally.